launch_asc.ks

```diff
@@ -7,92 +7,85 @@
+//	How far through our gravity turn are we? (0..1)
+	local gtPct is min(1,max(0, (ship:altitude - launch_gt0) / (launch_gt1 - launch_gt0))).
+//	Ideal gravity-turn azimuth (inclination) and facing at present altitude.
+	local pitch is arccos(gtPct).
 
-  // Ideal gravity-turn azimuth (inclination) and facing at present altitude.
-  local inclin is min(90, max(0, arccos(min(1,max(0,gtPct))))).
-  local gtFacing is heading ( hdglaunch, inclin) * r(0,0,180). //180 for shuttles, doesn't matter for rockets.
-
-  if gtPct <= 0 {
-    return heading (hdglaunch,90) + r(0,0,180). //Straight up.
-  } else {
-    return gtFacing.
-  }
+	return heading(hdglaunch, pitch) * r(0,0,launchRoll).
 }
 
 /////////////////////////////////////////////////////////////////////////////
 // Throttle function for continuous lock.
 /////////////////////////////////////////////////////////////////////////////
 
 function ascentThrottle {
-  // angle of attack
-  local aoa is vdot(ship:facing:vector, ship:velocity:surface).
-  // how far through the soup are we?
-  local atmPct is ship:altitude / (body:atm:height+1).
-  local spd is ship:airspeed.
-
-  // TODO adjust cutoff for atmospheric pressure; this works for kerbin
-  local cutoff is 200 + (400 * max(0, (atmPct*3))).
-
-  if spd > cutoff and launch_tSrbSep = 0 {
-    // going too fast during SRB ascent; avoid overheat or
-    // aerodynamic catastrophe by limiting throttle
-    return 1 - (1 * (spd - cutoff) / cutoff).
-  } else {
-    // Ease thottle when near the Apoapsis
-    local ApoPercent is ship:obt:apoapsis/apo.
-    local ApoCompensation is 0.
-    if ApoPercent > 0.9 set ApoCompensation to (ApoPercent - 0.9) * 10.
-    return 1.05 - min(1,max(0,ApoCompensation)).
-  }
+//	angle of attack
+	local aoa is vdot(ship:facing:vector, ship:velocity:surface).
+//	how far through the soup are we?
+	local atmPct is ship:altitude / (body:atm:height+1).
+	local spd is ship:airspeed.
+
+//	TODO: adjust cutoff for atmospheric pressure; this works for kerbin
+	local cutoff is 200 + (400 * max(0, (atmPct*3))).
+
+	if spd > cutoff {
+	//	going too fast - avoid overheat or aerodynamic catastrophe
+	//	by limiting throttle but not less than 10% to keep some gimbaling
+		return 1 - max(0.1, ((spd - cutoff) / cutoff)).
+	} else {
+	//	Ease throttle when near the Apoapsis
+		local ApoPercent is ship:obt:apoapsis/apo.
+		local ApoCompensation is 0.
+		if ApoPercent > 0.9 set ApoCompensation to (ApoPercent - 0.9) * 10.
+		return 1.05 - min(1,max(0,ApoCompensation)).
+	}
 }
 
 /////////////////////////////////////////////////////////////////////////////
-// Deploy fairings at proper altitude; call in a loop.
+// Deploy fairings and panels at proper altitude; call in a loop.
 /////////////////////////////////////////////////////////////////////////////
 
-function ascentFairing {
-  if ship:altitude > ship:body:atm:height {
-    if partsDeployFairings() uiBanner("ascend","Discard fairings").
-  }
+local deployed is false.
```

Comment:
Well, then we should make it the default parameter. BTW, I included the "noauto" tag mentioned.

Reply:
The default is "" because that will look at all untagged parts which represents the least surprise to other users of the function outside the launch script.  

If you build a vessel (engine, fuel tank, processor, antenna, panels, probe core, chute, etc) with no tags at all (a reasonable default), then `partsExtendSolarPanels(core:part:tag).` will extend all panels on the vessel.  That's the current behavior, so nobody gets a surprise.

Now build a vessel with a probe, where the parts involved in lifting (core, antenna, etc) all have the tag "Lifter", and the parts involved in the probe (core, antenna, etc) all have the tag "Probe".  If the core tagged "Lifter" is the one running the launch script, it will never deploy the probe's antenna or solar panel.  That's why it's important to have the argument in the script, but not as the default in `lib_parts.ks`.

With regard to staging, since I use the tags to identify the craft's mission as above I won't be using "noauto" as a tag option.  I also think that as currently written if you'd tagged the separator "noauto Probe" or "Probe noauto" then your `partDoEvent()` function may not act as you would expect if you tried to use it to decouple the separator.  I honestly don't know, though, I didn't test that bit.

Reply:
That is a thing I cannot agree with. It has to be default, because it is not only about launch script, it is about all the scripts, none would work, because default "" should not match Probe's panels, thus trying to call it without core's tag would only extend untagged panels, and that does not sound like a thing you want when running scripts from probe's core when you have tagged all the parts on it.

No surprise if you do not use tags at all. Surprise if you tag the core and all the parts (for the launch script) and then try to do anything from the core (as it won't do anything with tagged parts, only untagged and there are none such).

And about "noauto": actually, my launch script (and lib_staging) will work perfectly (see the note "noauto anything" or "noauto,more;whatever" is considered in my lib_staging), but I am afraid that `partsTagged` won't see it the same way. But, the thing is, that you do not need anything else on the separator, because it separates Lifter from Probe. Lifter will see it, Probe will not (because it won't be there at all when you need the probe's core). It makes things much much easier than your tagging everything, just add one "noauto" on separator between Lifter and Probe and you are done.

Reply:
I now understand why it works for you and doesn't for me.  It's well off-topic for this PR, but I'd be happy to share my code and craft with you so you can see what I'm talking about. :-)